src/main/java/org/apache/sling/engine/impl/SlingHttpServletResponseImpl.java

```diff
@@ -220,12 +225,12 @@ public void setStatus(final int sc, final String msg) {
 
     @Override
     public void reset() {
-        if (!this.isProtectHeadersOnInclude()) {
+        if (!this.isProtectHeadersOnInclude() || isError()) {
             super.reset();
         } else {
             // ignore if not committed
-            if (getResponse().isCommitted()) {
-                getResponse().reset();
+            if (super.isCommitted()) {
+                super.reset();
```

Comment:
The code did and still does something else than the comment says:
```suggestion
            if (!isCommitted()) {
                super.reset();
```

Reply:
reviews this with @cziegeler , check the new comment